tests/functional/b/bad_option_value_disable.py

```diff
@@ -0,0 +1,14 @@
+"""Tests for the disabling of bad-option-value."""
+# pylint: disable=invalid-name
+
+# pylint: disable=bad-option-value
+
+var = 1  # pylint: disable=a-removed-option
+
+# pylint: enable=bad-option-value
+
+var = 1  # pylint: disable=a-removed-option # [bad-option-value]
+
+# bad-option-value needs to be disabled before the bad option
+var = 1  # pylint: disable=a-removed-option, bad-option-value # [bad-option-value]
+var = 1  # pylint: disable=bad-option-value, a-removed-option
```

Comment:
My first instinct was "nice", but then I don't think it's instinctive, maybe it's better if it disable whatever the order ? After all for all other messages disabling it disable for the line and the order is not important.

Reply:
Hm, that would again be some additional code for very specific rare circumstances.

I think there is a very small group of users that will want to disable `bad-option-value` inline. Then there is an even smaller group who will make this mistake. I think that group is also likely to try to shuffle around the pragma's to fix the issue before opening an issue on the repo.
It's just a lot of additional code for something that I think can actually be "expected". Parsing pragma's in order is not something that is very out of the ordinary.

_My hesitance about this comes from the difficulty I had with this PR. You don't want to know how specific our current system is. Disable's on the first line counting for line zero but not for line two, disable's on function definitions also counting for decorators, etc. I'd rather simplify that make it more complicated again._

Reply:
Fair enough